packages/hono/src/index.ts

```diff
@@ -42,8 +49,8 @@ export type UnkeyConfig<TContext = Context> = {
   onError?: (c: TContext, err: UnkeyError) => Response | Promise<Response>;
 };
 
-export function unkey(config?: UnkeyConfig): MiddlewareHandler {
-  return async (c, next) => {
+export function unkey(config?: UnkeyConfig) {
+  return createMiddleware(async (c, next) => {
```

Comment:
use middleware helper to seamlessly interact with various platforms 

https://hono.dev/helpers/adapter#adapter-helper